tests/test_datasets/test_hierarchical_dataset.py

```diff
@@ -0,0 +1,169 @@
+            "segment": ["X"] * 2 + ["a"] * 2,
+            "target": [1, 2] + [10, 20],
+        }
+    )
+    df = TSDataset.to_dataset(df)
+    return df
+
+
+@pytest.fixture
+def diff_level_segments_df_exog():
+    df = pd.DataFrame(
+        {
+            "timestamp": ["2000-01-01", "2000-01-02"] * 2,
+            "segment": ["X"] * 2 + ["a"] * 2,
+            "exog": [1, 2] + [10, 20],
+        }
+    )
+    df = TSDataset.to_dataset(df)
+    return df
+
+
+@pytest.fixture
+def missing_segments_df():
+    df = pd.DataFrame(
+        {
+            "timestamp": ["2000-01-01", "2000-01-02"],
+            "segment": ["X"] * 2,
+            "target": [1, 2],
+        }
+    )
+    df = TSDataset.to_dataset(df)
+    return df
+
+
+@pytest.fixture
+def market_level_df():
+    df = pd.DataFrame(
+        {
+            "timestamp": ["2000-01-01", "2000-01-02"] * 2,
+            "segment": ["X"] * 2 + ["Y"] * 2,
+            "target": [1, 2] + [10, 20],
+        }
+    )
+    df = TSDataset.to_dataset(df)
+    return df
+
+
+@pytest.fixture
+def market_level_df_exog():
+    df_exog = pd.DataFrame(
+        {
+            "timestamp": ["2000-01-01", "2000-01-02"] * 2,
+            "segment": ["X"] * 2 + ["Y"] * 2,
+            "exog": [1, 2] + [10, 20],
+        }
+    )
+    df_exog = TSDataset.to_dataset(df_exog)
+    return df_exog
+
+
+@pytest.fixture
+def product_level_df():
+    df = pd.DataFrame(
+        {
+            "timestamp": ["2000-01-01", "2000-01-02"] * 4,
+            "segment": ["a"] * 2 + ["b"] * 2 + ["c"] * 2 + ["d"] * 2,
+            "target": [1, 2] + [10, 20] + [100, 200] + [1000, 2000],
+        }
+    )
+    df = TSDataset.to_dataset(df)
+    return df
+
+
+@pytest.fixture
+def simple_hierarchical_ts(market_level_df, hierarchical_structure):
+    df = market_level_df
+    ts = TSDataset(df=df, freq="D", hierarchical_structure=hierarchical_structure)
+    return ts
+
+
+def test_get_dataframe_level_diff_level_segments_fails(diff_level_segments_df, simple_hierarchical_ts):
+    with pytest.raises(ValueError, match="Segments in dataframe are from more than 1 hierarchical levels!"):
+        simple_hierarchical_ts._get_dataframe_level(df=diff_level_segments_df)
+
+
+def test_get_dataframe_level_missing_segments_fails(missing_segments_df, simple_hierarchical_ts):
+    with pytest.raises(ValueError, match="Some segments of hierarchical level are missing in dataframe!"):
+        simple_hierarchical_ts._get_dataframe_level(df=missing_segments_df)
+
+
+@pytest.mark.parametrize("df, expected_level", [("market_level_df", "market"), ("product_level_df", "product")])
+def test_get_dataframe(df, expected_level, simple_hierarchical_ts, request):
+    df = request.getfixturevalue(df)
+    df_level = simple_hierarchical_ts._get_dataframe_level(df=df)
+    assert df_level == expected_level
+
+
+def test_init_diff_level_segments_df_fails(diff_level_segments_df, hierarchical_structure):
```

Comment:
Don't we want to check the error `Some segments of hierarchical level are missing in dataframe` during init. As I understand, it can throw this error.